qa/rpc-tests/mempool_push.py

```diff
@@ -0,0 +1,136 @@
+        # Let's consume all BCH_UNCONF_DEPTH tx
+        self.nodes[0].set("mining.blockSize=8000000")
+
+        waitFor(DELAY_TIME, lambda: len(self.nodes[0].getblocktemplate()["transactions"])>=BCH_UNCONF_DEPTH)
+        blk3 = self.nodes[0].generate(1)[0]
+        blk3data = self.nodes[0].getblock(blk3)
+        # this would be ideal, but a particular block is not guaranteed to contain all tx in the mempool 
+        # assert_equal(len(blk3data["tx"]), BCH_UNCONF_DEPTH + 1)  # chain of BCH_UNCONF_DEPTH unconfirmed + coinbase
+        committedTxCount = len(blk3data["tx"])-1  # -1 to remove coinbase
+        waitFor(DELAY_TIME, lambda: self.nodes[1].getbestblockhash() == blk3)
+        waitFor(DELAY_TIME, lambda: self.nodes[1].getmempoolinfo()["size"] == unconfLeftOver - committedTxCount)
+        # make sure that everything that can be pushed is pushed
+        waitFor(DELAY_TIME, lambda: self.nodes[0].getmempoolinfo()["size"] == min(unconfLeftOver - committedTxCount, BCH_UNCONF_DEPTH))
+
+        # clean up: confirm all the left over tx from the prior test
+        self.nodes[0].set("mining.blockSize=8000000")
+        self.nodes[1].generate(1)
+
+        # Now let's run a more realistic test with 2 mining nodes of varying mempool depth, and one application node with a huge depth
+        logging.info("deep unconfirmed chain test")
+
+        # Because the TX push races the block, connect the network in a special way to avoid this race.
+        # This is undesirable for a test, but in the real network will likely result in a faster dispersal of the TX because the miners are interconnected
+        for n in self.nodes:
+            disconnect_all(n)
+        connect_nodes(self.nodes[0], 1)
+        connect_nodes(self.nodes[1], 2)
+
+        bal = self.nodes[2].getbalance()
+        addr = self.nodes[2].getnewaddress()
+
+        txhex = []
+        for i in range(0,100):
+          try:
+            txhex.append(self.nodes[2].sendtoaddress(addr, bal-1))  # enough so that it uses all UTXO, but has fee left over
+          except JSONRPCException as e: # an exception you don't catch is a testing error
+              print(str(e))
+              raise
+
+        count = 0
+        while self.nodes[2].getmempoolinfo()["size"] != 0:
+            # these checks aren't going to work at the end when I run out of tx so check for that
+            if self.nodes[2].getmempoolinfo()["size"] >= BCH_UNCONF_DEPTH*2:
+                waitFor(DELAY_TIME, lambda: self.nodes[0].getmempoolinfo()["size"] == BCH_UNCONF_DEPTH)
+                waitFor(DELAY_TIME, lambda: self.nodes[1].getmempoolinfo()["size"] == BCH_UNCONF_DEPTH*2)
+            logging.info("%d: sizes %d, %d, %d" % (count,self.nodes[0].getmempoolinfo()["size"],self.nodes[1].getmempoolinfo()["size"],self.nodes[2].getmempoolinfo()["size"]))
+            blk = self.nodes[0].generate(1)[0]
+            waitFor(DELAY_TIME, lambda: self.nodes[2].getbestblockhash() == blk)
+            count+=1
+
```

Comment:
what about a test for the decendant size limits?  is it possible here?

Reply:
As described in earlier comments, we should not and do not test based on descendant limits since that would be very broken.

Reply:
sorry, i meant to say anscestor size limits

Reply:
added a test using sendmany to create txes with a lot of data